### PR TITLE
#468: Add All Frontend Features

### LINKED ISSUE #468: MUI - Ensure that no frontend features were lost
### Description

the mui branch is really old, and although we've been merging develop into it, it's still psosible we lost some features coded on the frontened during all that.

### Acceptance Criteria

go through all the pull requests since the time the branch was open and make sure every feature exists

### Proposed Solution

.

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by anthonybernardi**
## Changes

I went through every single pr since the mui branch was opened and made sure all the changes were incorporated into this branch because we lost a few things in all the merges we did.

also i fixed a lot of stuff like the work package and project create forms

Closes #468

### CODE REVIEW

File: src/frontend/src/pages/CreateProjectPage/CreateProjectForm.tsx [hunk 1]
```diff
@@ -38,13 +40,9 @@ const CreateProjectForm: React.FC = () => {
 
   const handleCancel = () => history.goBack();
 
-  const redirectToCrTable = () => history.push(routes.CHANGE_REQUESTS);
-
   const handleSubmit = async (e: any) => {
```
Comment (Ari-S-123) [inline src/frontend/src/pages/CreateProjectPage/CreateProjectForm.tsx:43]: Should this be left as `any`?

File: src/frontend/src/pages/CreateWorkPackagePage/CreateWPFormView.tsx [hunk 2]
```diff
@@ -30,6 +28,7 @@ interface CreateWPFormViewProps {
   allowSubmit: boolean;
   onSubmit: (e: any) => void;
```
Comment (Ari-S-123) [inline src/frontend/src/pages/CreateWorkPackagePage/CreateWPFormView.tsx:29]: Here too.

File: src/frontend/src/utils/axios.ts [hunk 3]
```diff
@@ -11,8 +11,24 @@ axios.interceptors.response.use(
...
+    const message = error?.response?.data?.message;
+    if (message) {
+      throw new Error(message);
+    }
+
+    // this is how validation errors get sent
+    const errors = error?.response?.data?.errors;
+    if (errors) {
+      let messages = 'ERRORS:';
+      errors.forEach((element: any) => {
```
Comment (Ari-S-123) [inline src/frontend/src/utils/axios.ts:24]: And here.

**Review by Ari-S-123 — CHANGES_REQUESTED** [review]

File: src/frontend/src/pages/CreateWorkPackagePage/CreateWPFormView.tsx (hunk 2 above)
Comment (anthonybernardi) [reply src/frontend/src/pages/CreateWorkPackagePage/CreateWPFormView.tsx:29]: future ticket

File: src/frontend/src/pages/CreateProjectPage/CreateProjectForm.tsx (hunk 1 above)
Comment (anthonybernardi) [reply src/frontend/src/pages/CreateProjectPage/CreateProjectForm.tsx:43]: also will get changed whenever we move to react hook forms

File: src/frontend/src/utils/axios.ts (hunk 3 above)
Comment (anthonybernardi) [reply src/frontend/src/utils/axios.ts:24]: tru

File: src/frontend/src/pages/CreateWorkPackagePage/CreateWPFormView.tsx [hunk 4]
```diff
@@ -56,61 +57,62 @@ const CreateWPFormView: React.FC<CreateWPFormViewProps> = ({
...
               <DatePicker
                 label="Start Date"
                 inputFormat="yyyy-MM-dd"
-                value={startDateVal}
+                value={startDate}
                 onChange={(val) => {
-                  setStartDateVal(val);
-                  startDate(datePipe(val!));
+                  if (!val) return;
+                  setStartDate(val);
```
Comment (RChandler234) [inline src/frontend/src/pages/CreateWorkPackagePage/CreateWPFormView.tsx:104]: this should be in a small helper method

**Review by RChandler234 — CHANGES_REQUESTED** [review]

**Review by RChandler234 — APPROVED** [review]
LGTM 🔥

**Review by Ari-S-123 — APPROVED** [review]

**Pull request merged**